stubs/toml/toml/decoder.pyi

```diff
@@ -0,0 +1,57 @@
+import sys
+from _typeshed import StrPath, SupportsRead
+from typing import Any, Callable, Generic, MutableMapping, Pattern, Text, TypeVar
+
+_T = TypeVar("_T")
+
+if sys.version_info >= (3, 3):
+    FNFError = FileNotFoundError
+else:
+    FNFError = IOError
+
+TIME_RE: Pattern[str]
+
+class TomlDecodeError(ValueError):
+    msg: str
+    doc: str
+    pos: int
+    lineno: int
+    colno: int
+    def __init__(self, msg: str, doc: str, pos: int) -> None: ...
+
+class CommentValue(Generic[_T]):
+    val: _T
+    comment: str
+    def __init__(self, val: _T, comment: str, beginline: bool, _dict: type[MutableMapping[str, _T]]) -> None: ...
+    def __getitem__(self, key: _T) -> _T: ...
+    def __setitem__(self, key: _T, value: _T) -> None: ...
+    def dump(self, dump_value_func: Callable[[_T], str]) -> str: ...
+
+def load(
+    f: StrPath | list[Text] | SupportsRead[str], _dict: type[MutableMapping[str, _T]] = ..., decoder: TomlDecoder[_T] | None = ...
```

Comment:
I looked at toml's source code and I think this is what `f`'s type should be
```
load(f, ...)
    isnstance(f, (bytes, str, PurePath)               # so bytes | str | PurePath
        open(f)
    isinstance(f, list)                               # list of what is common between os.path.exists and load
        load(t, ...) for t in f if os.path.exists(t)  # so str, bytes or PurePath since exists has PathLike[str]
    loads(f.read())                                   # and SupportsRead[Text]
```
so that would be
```python
if sys.version_info >= (3, 4):
    from pathlib import PurePath

    _PathLike = str | bytes | PurePath
else:
    _PathLike = str | bytes

def load(f: _PathLike | list[_PathLike] | SupportsRead[Text], ...)```